[2019-10-04 19:54:59,543] {taskinstance.py:620} INFO - Dependencies all met for <TaskInstance: capstone_project_dag.Run_final_tables_quality_checks 2019-10-05T00:44:32.721770+00:00 [queued]>
[2019-10-04 19:54:59,552] {taskinstance.py:620} INFO - Dependencies all met for <TaskInstance: capstone_project_dag.Run_final_tables_quality_checks 2019-10-05T00:44:32.721770+00:00 [queued]>
[2019-10-04 19:54:59,552] {taskinstance.py:838} INFO - 
--------------------------------------------------------------------------------
[2019-10-04 19:54:59,552] {taskinstance.py:839} INFO - Starting attempt 2 of 4
[2019-10-04 19:54:59,552] {taskinstance.py:840} INFO - 
--------------------------------------------------------------------------------
[2019-10-04 19:54:59,571] {taskinstance.py:859} INFO - Executing <Task(DataQualityCheckOperator): Run_final_tables_quality_checks> on 2019-10-05T00:44:32.721770+00:00
[2019-10-04 19:54:59,572] {base_task_runner.py:133} INFO - Running: ['airflow', 'run', 'capstone_project_dag', 'Run_final_tables_quality_checks', '2019-10-05T00:44:32.721770+00:00', '--job_id', '1016', '--pool', 'default_pool', '--raw', '-sd', 'DAGS_FOLDER/project_dag.py', '--cfg_path', '/var/folders/dr/9yvjlnm11xv25k2gknf2n3z00000gn/T/tmpx0fophyf']
[2019-10-04 19:55:00,121] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks /usr/local/lib/python3.7/site-packages/airflow/configuration.py:627: DeprecationWarning: You have two airflow.cfg files: /Users/victorialemay/airflow/airflow.cfg and /Users/victorialemay/Courses/Udacity/Data Engineering Nanodegree/capstone/project-archive/airflow/airflow.cfg. Airflow used to look at ~/airflow/airflow.cfg, even when AIRFLOW_HOME was set to a different value. Airflow will now only read /Users/victorialemay/Courses/Udacity/Data Engineering Nanodegree/capstone/project-archive/airflow/airflow.cfg, and you should remove the other file
[2019-10-04 19:55:00,121] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks   category=DeprecationWarning,
[2019-10-04 19:55:00,219] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks [2019-10-04 19:55:00,219] {settings.py:213} INFO - settings.configure_orm(): Using pool settings. pool_size=5, max_overflow=10, pool_recycle=1800, pid=1119
[2019-10-04 19:55:00,446] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks [2019-10-04 19:55:00,445] {__init__.py:51} INFO - Using executor LocalExecutor
[2019-10-04 19:55:00,903] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks [2019-10-04 19:55:00,902] {dagbag.py:90} INFO - Filling up the DagBag from /Users/victorialemay/Courses/Udacity/Data Engineering Nanodegree/capstone/project-archive/airflow/dags/project_dag.py
[2019-10-04 19:55:00,945] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks [2019-10-04 19:55:00,945] {cli.py:516} INFO - Running <TaskInstance: capstone_project_dag.Run_final_tables_quality_checks 2019-10-05T00:44:32.721770+00:00 [running]> on host 1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.ip6.arpa
[2019-10-04 19:55:00,978] {data_quality_check.py:42} INFO - Checking data in port_id column of i94_immigration table
[2019-10-04 19:55:00,992] {logging_mixin.py:95} INFO - [[34m2019-10-04 19:55:00,991[0m] {[34mbase_hook.py:[0m84} INFO[0m - Using connection to: [1mid: redshift. Host: redshift-cluster-1.c4i0rfzboubh.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: {}[0m[0m
[2019-10-04 19:55:01,758] {data_quality_check.py:51} INFO - Found records in i94_immigration
[2019-10-04 19:55:01,762] {data_quality_check.py:42} INFO - Checking data in i94_port_code column of port_codes table
[2019-10-04 19:55:01,776] {logging_mixin.py:95} INFO - [[34m2019-10-04 19:55:01,775[0m] {[34mbase_hook.py:[0m84} INFO[0m - Using connection to: [1mid: redshift. Host: redshift-cluster-1.c4i0rfzboubh.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: {}[0m[0m
[2019-10-04 19:55:02,499] {data_quality_check.py:51} INFO - Found records in port_codes
[2019-10-04 19:55:02,499] {data_quality_check.py:42} INFO - Checking data in date column of land_temperatures table
[2019-10-04 19:55:02,519] {logging_mixin.py:95} INFO - [[34m2019-10-04 19:55:02,519[0m] {[34mbase_hook.py:[0m84} INFO[0m - Using connection to: [1mid: redshift. Host: redshift-cluster-1.c4i0rfzboubh.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: {}[0m[0m
[2019-10-04 19:55:03,239] {data_quality_check.py:51} INFO - Found records in land_temperatures
[2019-10-04 19:55:03,240] {data_quality_check.py:42} INFO - Checking data in port_code_id column of us_city_demographics table
[2019-10-04 19:55:03,252] {logging_mixin.py:95} INFO - [[34m2019-10-04 19:55:03,252[0m] {[34mbase_hook.py:[0m84} INFO[0m - Using connection to: [1mid: redshift. Host: redshift-cluster-1.c4i0rfzboubh.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: {}[0m[0m
[2019-10-04 19:55:03,973] {data_quality_check.py:51} INFO - Found records in us_city_demographics
[2019-10-04 19:55:03,973] {data_quality_check.py:30} INFO - Checking data in port_id column of i94_immigration table
[2019-10-04 19:55:03,987] {logging_mixin.py:95} INFO - [[34m2019-10-04 19:55:03,987[0m] {[34mbase_hook.py:[0m84} INFO[0m - Using connection to: [1mid: redshift. Host: redshift-cluster-1.c4i0rfzboubh.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: {}[0m[0m
[2019-10-04 19:55:04,712] {data_quality_check.py:35} INFO - Found [(0,)] NULL records
[2019-10-04 19:55:04,712] {taskinstance.py:1051} ERROR - NULL records were found for the port_id column
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 926, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/Users/victorialemay/Courses/Udacity/Data Engineering Nanodegree/capstone/project-archive/airflow/plugins/operators/data_quality_check.py", line 24, in execute
    getattr(self, check_function)(redshift_hook)
  File "/Users/victorialemay/Courses/Udacity/Data Engineering Nanodegree/capstone/project-archive/airflow/plugins/operators/data_quality_check.py", line 36, in has_nulls
    raise ValueError("NULL records were found for the {} column".format(column))
ValueError: NULL records were found for the port_id column
[2019-10-04 19:55:04,717] {taskinstance.py:1074} INFO - Marking task as UP_FOR_RETRY
[2019-10-04 19:55:04,741] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks Traceback (most recent call last):
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks   File "/usr/local/bin/airflow", line 32, in <module>
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks     args.func(args)
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks   File "/usr/local/lib/python3.7/site-packages/airflow/utils/cli.py", line 74, in wrapper
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks     return f(*args, **kwargs)
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks   File "/usr/local/lib/python3.7/site-packages/airflow/bin/cli.py", line 522, in run
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks     _run(args, dag, ti)
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks   File "/usr/local/lib/python3.7/site-packages/airflow/bin/cli.py", line 440, in _run
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks     pool=args.pool,
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks   File "/usr/local/lib/python3.7/site-packages/airflow/utils/db.py", line 74, in wrapper
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks     return func(*args, **kwargs)
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks   File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 926, in _run_raw_task
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks     result = task_copy.execute(context=context)
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks   File "/Users/victorialemay/Courses/Udacity/Data Engineering Nanodegree/capstone/project-archive/airflow/plugins/operators/data_quality_check.py", line 24, in execute
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks     getattr(self, check_function)(redshift_hook)
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks   File "/Users/victorialemay/Courses/Udacity/Data Engineering Nanodegree/capstone/project-archive/airflow/plugins/operators/data_quality_check.py", line 36, in has_nulls
[2019-10-04 19:55:04,742] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks     raise ValueError("NULL records were found for the {} column".format(column))
[2019-10-04 19:55:04,743] {base_task_runner.py:115} INFO - Job 1016: Subtask Run_final_tables_quality_checks ValueError: NULL records were found for the port_id column
[2019-10-04 19:55:09,540] {logging_mixin.py:95} INFO - [[34m2019-10-04 19:55:09,540[0m] {[34mlocal_task_job.py:[0m105} INFO[0m - Task exited with return code 1[0m
